java/src/main/java/org/digma/intellij/plugin/idea/psi/java/AbsJaxrsFramework.java

```diff
@@ -114,59 +112,102 @@ private boolean isJaxRsHttpRelevant() {
         return jaxrsPathAnnotationClass != null;
     }
 
+    /**
+     * Overriding since have different logic,
+     * in this impl searching for classes/methods in file and see if they inherit from super classes which declare on @PATH annotations.
+     */
     @Override
-    public void endpointDiscovery(@NotNull PsiFile psiFile, @NotNull DocumentInfo documentInfo) {
+    public List<EndpointInfo> lookForEndpoints(@NotNull PsiFile psiFile) {
         lateInit();
         if (!isJaxRsHttpRelevant()) {
-            return;
+            return Collections.emptyList();
         }
 
-        var psiFacade = JavaPsiFacade.getInstance(project);
-        final PsiClass appPathAnnotationClass = psiFacade.findClass(getApplicationPathAnnotationClassFqn(), GlobalSearchScope.allScope(project));
-
-        List<PsiClass> allClassesInFile = JavaPsiUtils.getClassesWithin(psiFile);
-        for (PsiClass currClass : allClassesInFile) {
-            final PsiAnnotation controllerPathAnnotation = JavaPsiUtils.findNearestAnnotation(currClass, JAX_RS_PATH_ANNOTATION_STR());
+        Collection<PsiClass> allClassesInFile = PsiTreeUtil.findChildrenOfType(psiFile, PsiClass.class);
```

Comment:
direct psi api calls have better performance then finders.
so psiFile.getClasses is better the PsiTreeUtil.findChildrenOfType

Reply:
PsiJavaFile.getClasses was not working well

Reply:
That is strange. but i can't argue. usually PsiJavaFile.getClasses works well unless index is not ready or corrupted.